### ISSUE TITLE
Call to avinor/storage-account/azurerm uses existing resource group. Terraform cannot apply from clean tfstate.

### ISSUE THREAD

**Issue by ghouser** [avinor/terraform-azurerm-virtual-network-hub#14]
See issue https://github.com/avinor/terraform-azurerm-storage-account/issues/16

`avinor/storage-account/azurerm` creates a new resource group, but the current behavior passes an existing resource group name:

https://github.com/avinor/terraform-azurerm-virtual-network-hub/blob/cdb26e88cd9dc39f99089e266a31209f1d6bda99/main.tf#L276
```
module "storage" {
  source  = "avinor/storage-account/azurerm"
  version = "3.0.1"

  name                = var.name
  resource_group_name = azurerm_resource_group.vnet.name
```

The result is `storage` attempts to create a resource group that already exists. From terraform plan:

```
# module.datacenter-hub.azurerm_resource_group.vnet will be created
  + resource "azurerm_resource_group" "vnet" {
      + id       = (known after apply)
      + location = "eastus"
      + name     = "qa-ue"
    }
    
    # module.datacenter-hub.module.storage.azurerm_resource_group.storage will be created
  + resource "azurerm_resource_group" "storage" {
      + id       = (known after apply)
      + location = "eastus"
      + name     = "qa-ue"
    }
 ```

The result is I can only apply from an existing TF state, where the vnet group is already created and imported. I'm trying to build such that I can terraform apply from a completely empty tfstate.

**Comment by spatankar23** [comment]
Any update or solution for above issue.

**Comment by spatankar23** [comment]
There exists another issue with fw-diag in datacenter-hub/main.tf (line no: 621 - 640)

Both the reported issues can be fixed as per attached images.

Issue 1
![image](https://user-images.githubusercontent.com/90892887/159421312-0f3a881b-2b00-41bc-90c5-6523d8579c93.png)

Issue 2
![image](https://user-images.githubusercontent.com/90892887/159420811-7695010e-d48f-406f-8ee0-248f0bb10c6c.png)

(The update made in main.tf from fw-diag to fw-diag-hub as per attached image resolves the import issue. Expecting similar fix)

The result is I can only apply from an existing TF state, where the fw-diag is already created and imported. I'm trying to build such that I can terraform apply from a completely empty tfstate.

**Comment by yngveh** [comment]
This should be fix in release v3.0.1 of the module and creating from a clean state is now working as expected.

NB! Take care when upgrading from an old state! Se  README.md for details.
